src/sql/SqlService.ts

```diff
@@ -0,0 +1,290 @@
+                response.updateCount
+            );
+        }
+    }
+
+    /**
+     * Validates SqlStatement
+     *
+     * @param sqlStatement
+     * @throws RangeError if validation is not successful
+     * @internal
+     */
+    static validateSqlStatement(sqlStatement: SqlStatement | null): void {
+        if (sqlStatement === null) {
+            throw new RangeError('Sql statement cannot be null');
+        }
+        tryGetString(sqlStatement.sql);
+        if (sqlStatement.hasOwnProperty('params')) // if params is provided, validate it
+            tryGetArray(sqlStatement.params);
+        if (sqlStatement.hasOwnProperty('options')) // if options is provided, validate it
+            SqlServiceImpl.validateSqlStatementOptions(sqlStatement.options);
+    }
+
+    /**
+     * Validates SqlStatementOptions
+     *
+     * @param sqlStatementOptions
+     * @throws RangeError if validation is not successful
+     * @internal
+     */
+
+    static validateSqlStatementOptions(sqlStatementOptions: SqlStatementOptions): void {
+        if (sqlStatementOptions.hasOwnProperty('schema'))
+            tryGetString(sqlStatementOptions.schema);
+
+        if (sqlStatementOptions.hasOwnProperty('timeoutMillis')) {
+            const longValue = tryGetLong(sqlStatementOptions.timeoutMillis);
+            if (longValue.lessThanOrEqual(Long.fromInt(-2))) {
+                throw new RangeError('Timeout millis cannot be less than -1');
+            }
+        }
+
+        if (sqlStatementOptions.hasOwnProperty('cursorBufferSize') && tryGetNumber(sqlStatementOptions.cursorBufferSize) <= 0) {
+            throw new RangeError('Cursor buffer size cannot be negative');
+        }
+
+        if (sqlStatementOptions.hasOwnProperty('expectedResultType'))
+            tryGetEnum(SqlExpectedResultType, sqlStatementOptions.expectedResultType);
+
+        if (sqlStatementOptions.hasOwnProperty('returnRawResult'))
+            tryGetBoolean(sqlStatementOptions.returnRawResult);
+    }
+
+    /**
+     * Converts datetime related wrapper classes to string to be able to serialize them. (No default serializers yet)
+     * @internal
+     * @param value
+     */
+    convertToStringIfDatetimeValue(value: any) {
+        if (value instanceof HzLocalTime || value instanceof HzLocalDate || value instanceof HzLocalDateTime) {
+            return value.toString();
+        } else if (value instanceof HzOffsetDateTime) {
+            return value.toISOString();
+        } else {
+            return value;
+        }
+    }
+
+    execute(sql: SqlStatement): SqlResult;
+    execute(sql: string, params?: any[], options?: SqlStatementOptions): SqlResult;
+    execute(sql: string | SqlStatement, params?: any[], options?: SqlStatementOptions): SqlResult {
+        let sqlStatement: SqlStatement;
+
+        if (typeof sql === 'string') {
+            sqlStatement = {
+                sql: sql
+            };
+            if (params !== undefined) { // todo: document this
```

Comment:
Will you deal with this todo?

Reply:
I was going to write a comment, I wrote now